src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1593,6 +1595,23 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         RsDiagnostic.AwaitOutsideAsyncContext(element, fix).addToHolder(holder)
     }
 
+    private fun checkTypeParameterList(holder: RsAnnotationHolder, typeParameterList: RsTypeParameterList) {
```

Comment:
Hmm, let's try a reverse approach - instead of handling `RsTypeParameterList` and recursing into nested types we could handle `RsPathType` and check its ancestors